image-processing/UI.py

```diff
@@ -1,484 +1,202 @@
-
-                        if not file_path.exists():
-                            print("Sorry, this file does not exist. Please re-enter including any spaces, and the file extension, e.g. .jpeg")
-                            continue
-
-                        else:
-                            while True:
-                                grouping = input("is the file '{}' a scan of a low, or high areal weight sample? ".format(filename_input)).lower()
-                                if not (grouping == 'high' or grouping =='low'):
-                                    print('response not recognised, please respond with either high or low')
-                                    continue
-                                break
-                            file = np.load(file_path, allow_pickle = True)
-                            paths_saved.append(file_path)
-                            scans_saved.append([file,grouping])
-
-                            print("File added successfully.")
-                        no_file = True
-                            # print(paths_saved)
-
-
-                print("Please respond 'yes' or 'no' to the following questions:")
-
-                while True:
-                    view_plot = input("Would you like to view the output plot from the analysis software? ").lower()
-
-                    if not view_plot in ('yes', 'no'):
-                        yesno_error()
-                        continue
-
-                    break
-
-                while True:
-                    set_parameters = input("Would you like to use preset values or set your own parameters? ('yes' for presets, 'no' for custom parameters) ").lower()
-
-                    if not set_parameters in ('yes', 'no'):
-                        yesno_error()
-                        continue
-
-                    break
-
-
-                for scan in range(len(scans_saved)):
-
-                    if set_parameters == 'yes':
+    def input_sample(self):
+        sample_type = input("Type of Sample: ")
+        
+        if sample_type in [str(n) for n in range(len(self.material_types) + 1)]:
+            print("Please ensure samples are scanned at 75dpi, or the next closest possible resolution.\n")
+            return int(sample_type)
 
-                            original = scans_saved[scan][0]
-                            sigma = 1
-                            row = 250
-                            finder = linefinder(original, sigma, row)
+        # elif sample_type.lower() == "back":
+        #     self.get_path()
+            
+        else:
+            print(f"Ensure an integer between 1 and {len(self.material_types)} inclusive is entered, please try again.\n")
+            self.input_sample()
+    
+    def input_grouping(self):
+        grouping = input(f"Does the folder contain scans of low or high areal weight sample? Enter 'low' or 'high': ").lower()
+        if grouping in ('high', 'low'):
```

Comment:
I'll try to add 'all' or 'help' in here too

Reply:
done (TF)